.github/workflows/macos-java.yml

```diff
@@ -49,9 +44,7 @@ jobs:
       - name: bootstrap
         run: |
           sed -i '' 's/-undefined suppress//g' configure.ac
```

Comment:
What is this doing and why? It seems to originate from @ddeclerck.
We should at least add a comment there.

Reply:
"Recent" versions of XCode (>= 15 ?) include a new linker, which does not support the `-undefined` flag. This substitution removes that option, which would otherwise cause the testsuite to fail on MacOS. Until a definitive solution is found...

Reply:
I guess this would need to be checked by `configure`.

Reply:
configure uses this only in a single place - "newer" MacOS and a compiler claiming to be GCC:

```sh
case $host_os in
  rhapsody*)
	if test "$COB_USES_GCC" = yes; then
		COB_SHARED_OPT="-bundle -flat_namespace -undefined suppress"
	fi
	;;
```
... so I've learned something new: https://en.wikipedia.org/wiki/Rhapsody_(operating_system)

I've also checked the used files from https://git.savannah.gnu.org/cgit/config.git and will update some "soon" but that won't help for this specific issue.

For that we can add a separate case entry (what is `$host_os` in the failing case?) or try if we can link successfully w/wo those options.

Reply:
The `libtool.m4` seems to distinguish using darwin versions as well:
```m4
# serial 58 LT_INIT
…
    case $host_os in
    rhapsody* | darwin1.[[012]])
      _lt_dar_allow_undefined='$wl-undefined ${wl}suppress' ;;
    darwin1.*)
      _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
    darwin*) # darwin 5.x on
      # if running on 10.5 or later, the deployment target defaults
      # to the OS version, if on x86, and 10.4, the deployment
      # target defaults to 10.4. Don't you love it?
      case ${MACOSX_DEPLOYMENT_TARGET-10.0},$host in
	10.0,*86*-darwin8*|10.0,*-darwin[[912]]*)
	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
	10.[[012]][[,.]]*)
	  _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
	10.*|11.*)
	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
      esac
    ;;
  esac
…
```
(not sure that's relevant though)